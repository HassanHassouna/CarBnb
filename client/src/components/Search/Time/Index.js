import React, { forwardRef } from "react"
import MenuItem from "@mui/material/MenuItem"
import FormControl from "@mui/material/FormControl"
import Select from "@mui/material/Select"
import { makeStyles } from "@material-ui/core/styles"
import { possibleTimes } from "./data"
import "./style.css"

const useStyles = makeStyles(() => ({
  notchedOutline: {
    border: "none",
  },
  select: {
    "& li": {
      fontSize: 12,
    },
  },
  dropdownStyle: {
    backgroundColor: "lightgrey",
  },
}))

const Time = forwardRef((props, ref) => {
  const classes = useStyles()
  const [time, setTime] = React.useState("")
  const handleChange = (event) => {
    setTime(event.target.value)
  }

  return (
    <FormControl sx={{ minWidth: 125, mr: 1 }}>
      <Select
        variant="standard"
        MenuProps={{ classes: { paper: classes.dropdownStyle } }}
        value={time}
        onChange={handleChange}
        inputRef={ref}
        inputProps={{ border: "sildpink" }}
        required
      >
        {possibleTimes.map((time, index) => (
          <MenuItem key={index} value={time}>
            {time}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  )
})

export default Time
