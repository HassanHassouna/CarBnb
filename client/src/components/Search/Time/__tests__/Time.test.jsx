import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Time from '../Index';

describe('Time', () => {
  test('should render with no parameters', () => {
    const { time } = render(<Time />);

    expect(time).toMatchSnapshot();
  });
});
