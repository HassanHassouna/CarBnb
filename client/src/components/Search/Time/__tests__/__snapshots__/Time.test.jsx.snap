// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time should render with no parameters 1`] = `undefined`;
